client/src/main/java/io/pravega/client/netty/impl/ClientConnectionImpl.java

```diff
@@ -50,9 +55,12 @@ public void send(WireCommand cmd) throws ConnectionFailedException {
 
     @Override
     public void send(Append append) throws ConnectionFailedException {
+        Timer timer = new Timer();
         checkClientConnectionClosed();
         nettyHandler.setRecentMessage();
         Futures.getAndHandleExceptions(nettyHandler.getChannel().writeAndFlush(append), ConnectionFailedException::new);
+        nettyHandler.getUpdateMetric().updateSuccessMetric(metricKey(ClientMetricNames.CLIENT_APPEND_LATENCY, segmentTags(append.getSegment())),
```

Comment:
Perhaps we can pass the tags into the function so that any string concatenation happens inside ClientMetricUpdater in the background thread.